tests/flows/test_vesting.py

```diff
@@ -106,73 +103,535 @@ def test_main(
+            erc20_token.address
+        )
+        assert (
+            available_vesting_after > available_vesting_before
+        ), "Available vesting was not increased!"
+        amount_vested_so_far = side_staking.get_vesting_amount_so_far(
+            erc20_token.address
+        )
+        tx_hash = side_staking.get_vesting(erc20_token.address, publisher_wallet)
+
+        # counter is used to count the number of times when get_vesting is called
+        # (it mints another block that change the formula)
+        counter += 1
+        vested_amount = side_staking.get_amount_vested_from_event(
+            tx_hash=tx_hash, erc20_token=erc20_token, from_wallet=publisher_wallet
+        )
+        target_vesting_amount = (
+            int(((block + 1 + counter) * vesting_amount) / vested_blocks)
+            - amount_vested_so_far
+        )
+        assert vested_amount == target_vesting_amount
+        # claim tokens after vested blocks
+        pmt_collector = erc20_token.get_payment_collector()
+        assert dt_balance_before < erc20_token.balanceOf(pmt_collector)
+        assert (
+            erc20_token.balanceOf(pmt_collector) == vested_amount + amount_vested_so_far
+        )
+
 
+def test_vesting_publisher_exit_scam(
+    web3,
+    config,
+    publisher_wallet,
+    consumer_wallet,
+    another_consumer_wallet,
+    factory_router,
+    side_staking,
+):
+    """Tests publisher exit scam flow when the owner of the pool claimed the datatokens from vesting,
+    waits for others to add liquidity in his pool and then dumps the pool."""
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, publisher_wallet, publisher_wallet, cap=to_wei(2000)
+    )
+
+    mint_fake_OCEAN(config)
+    vesting_amount = to_wei(18)
+    initial_ocean_liq = to_wei(200)
+    pmt_collector = erc20_token.get_payment_collector()
+
+    ocean_contract = ERC20Token(web3=web3, address=get_address_of_type(config, "Ocean"))
+
+    bpool = _deploy_ocean_pool(
+        erc20_token=erc20_token,
+        factory_router=factory_router,
+        config=config,
+        web3=web3,
+        from_wallet=publisher_wallet,
+    )
+
+    # check if the vesting amount is correct
+    assert side_staking.get_vesting_amount(erc20_token.address) == vesting_amount
+
+    # check if vesting is correct
+    dt_balance_before = erc20_token.balanceOf(pmt_collector)
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+
+    # advance 10 blocks to see if available vesting increased
+    for _ in range(9):
+        # send dummy transactions to increase block count (not part of the actual functionality)
+        send_ether(consumer_wallet, ZERO_ADDRESS, 1)
+
+    available_vesting_after = side_staking.get_available_vesting(erc20_token.address)
+    assert (
+        available_vesting_after > available_vesting_before
+    ), "Available vesting was not increased!"
```

Comment:
You don't need to test this in this function, since you've tested this basic functionality already in other functions. Delete lines.

Focus on getting to the heart of what _this_ function is trying to test for. Cut the rest out.

Reply:
Kept one of this in `test_vesting_main_flow` & cut the rest out